regparser/tree/depth/derive.py

```diff
@@ -142,6 +147,23 @@ def derive_depths(original_markers, additional_constraints=[]):
     for assignment in problem.getSolutionIter():
         assignment = _decompress_markerless(assignment, original_markers)
         solutions.append(Solution(assignment))
+
+    if not solutions and not relaxed_constraints:
+        deemphasized_markers = []
+        logging.warning("Could not derive paragraph depths. Retrying with relaxed constraints.")
+        for marker in original_markers:
+            emphasized = re.match(r"<E .*>(.*)</E>", marker)
+            if emphasized:
+                deemphasized_markers.append(emphasized.groups()[0])
+            else:
+                deemphasized_markers.append(marker)
+       
+        solutions = derive_depths(deemphasized_markers, [optional_rules.star_new_level,
+            optional_rules.limit_paragraph_types(
+                markers.lower, markers.upper, markers.ints, markers.roman,
+                markers.em_ints, markers.em_roman, markers.stars,
+                markers.markerless)], relaxed_constraints=True)
```

Comment:
This is probably the wrong place to perform this retry logic; I think you'd want it around here instead: https://github.com/18F/regulations-parser/blob/master/regparser/tree/xml_parser/paragraph_processor.py#L124

This would also allow you to swap out _active_ optional rules rather than have a fixed list here.